org.codeblocks.codeblocks.json

```diff
@@ -0,0 +1,106 @@
+{
+    "app-id": "org.codeblocks.codeblocks",
+    "runtime": "org.freedesktop.Sdk",
+    "runtime-version": "1.6",
+    "sdk": "org.freedesktop.Sdk",
+    "rename-appdata-file": "codeblocks.appdata.xml",
+    "rename-desktop-file": "codeblocks.desktop",
+    "rename-icon": "codeblocks",
+    "command": "codeblocks",
+    "finish-args": [
+        "--allow=devel",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--share=network",
+        "--filesystem=home"
+    ],
+    "cleanup": [
+        "/include",
+        "/lib/pkgconfig",
+        "/share/aclocal",
+        "/share/doc",
+        "/share/man",
+        "*.a"
+    ],
+    "modules": [
+        {
+            "name": "libXmu",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://www.x.org/releases/individual/lib/libXmu-1.1.2.tar.gz",
+                    "sha256": "e5fd4bacef068f9509b8226017205040e38d3fba8d2de55037200e7176c13dba"
+                }
+            ]
+        },
+        {
+            "name": "libXaw",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://www.x.org/releases/individual/lib/libXaw-1.0.13.tar.gz",
+                    "sha256": "7e74ac3e5f67def549722ff0333d6e6276b8becd9d89615cda011e71238ab694"
+                }
+            ]
+        },
+        {
+            "name": "xterm",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://invisible-mirror.net/archives/xterm/xterm-331.tgz",
+                    "sha256": "9ae856a30fd93046be93952a6898ba47f6f88ad6a988a7c949c4c80d5199ef10"
+                }
+            ]
+        },
+	{
+	    "name": "wxWidgets",
+	    "rm-configure": true,
+            "cleanup": [
+                "/bin",
+                "/lib/wx",
+                "/share/bakefile"
+            ],
+	    "sources": [
+		{
+		    "type": "archive",
+		    "url": "https://github.com/wxWidgets/wxWidgets/releases/download/v3.0.4/wxWidgets-3.0.4.tar.bz2",
```

Comment:
Requires testing but a lot of apps using `3.1.x` and IME it fixes a lot of problems with Gtk3 support.

Reply:
I actually tested with both 3.0.4 and 3.1, with both HiDpi and non HiDpi displays, and in this case 3.0.4 produced way better results.